### PR TITLE
Make ibis server be a mixed rust/python project

### PR DESCRIPTION
**Pull request by grieve54706**
To import rust modeling core into Python.

Ideally, provide manifest builder to Python. But we are not ready. We still adjust the builder in Rust.

We provide the function with string argument for Python now.

```python
from wren_engine import modeling_core

manifest_base64_str = base64.b64encode(orjson.dumps(manifest)).decode("utf-8")
rewritten_sql = modeling_core.transform_sql(manifest_base64_str, sql)
```

### CODE REVIEW

File: ibis-server/README.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Ibis server
 Ibis server is a server base on Python that provides the API for rewriting the queries with [Modeling Definition Language (MDL)](https://docs.getwren.ai/engine/concept/what_is_mdl) manifest and connect data source via [Ibis](https://github.com/ibis-project/ibis). It is built on top of the [FastAPI](https://github.com/tiangolo/fastapi) framework. \
-We still need to run the Java engine for rewriting the queries. In the future, we will rewrite the modeling core of java engine to [Rust](https://github.com/rust-lang/rust) and integrate the modeling core into the Ibis server.
+We still need to run the Java engine for rewriting the queries. In the future, we will rewrite the modeling core of java engine to [Rust](https://github.com/rust-lang/rust) and integrate the modeling core into the Ibis server. \
```
Comment (goldmedal) [inline ibis-server/README.md:3]:
```suggestion
We still need to run the Java engine for rewriting the queries. In the future, we will redesign the modeling core of java engine to [Rust](https://github.com/rust-lang/rust) and integrate the modeling core into the Ibis server. \
```

File: ibis-server/rust/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            ]
+        }"#;
+        let v: Value = serde_json::from_str(data).unwrap();
+        let mdl_base64: String = BASE64_STANDARD.encode(v.to_string().as_bytes());
+        let transformed_sql = transform_sql(
+            &mdl_base64,
+            "SELECT * FROM my_catalog.my_schema.customer",
+        )
+        .unwrap();
+        assert_eq!(transformed_sql, "SELECT \"my_catalog\".\"my_schema\".\"customer\".\"custkey\", \"my_catalog\".\"my_schema\".\"customer\".\"name\" FROM (SELECT \"customer\".\"c_custkey\" AS \"custkey\", \"customer\".\"c_name\" AS \"name\" FROM \"customer\") AS \"customer\"");
```
Comment (goldmedal) [inline ibis-server/rust/src/lib.rs:66]: How about use rust string literal, then we don't need the escape `\` char.
```suggestion
        assert_eq!(transformed_sql, r#"SELECT \"my_catalog\".\"my_schema\".\"customer\".\"custkey\", \"my_catalog\".\"my_schema\".\"customer\".\"name\" FROM (SELECT \"customer\".\"c_custkey\" AS \"custkey\", \"customer\".\"c_name\" AS \"name\" FROM \"customer\") AS \"customer\""#);
```

File: .github/workflows/ibis-ci.yml [hunk 3]
```diff
@@ -47,6 +47,21 @@ jobs:
...
+        uses: actions/cache@v3
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            ibis-server/rust/target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('ibis-server/rust/Cargo.lock') }}
+      - name: Test rust code
```
Comment (goldmedal) [inline .github/workflows/ibis-ci.yml:60]: We need to have test for different platform (macos, mac-arm, linux, windows) but I think we don't need to do that in this PR. Have another follow up PR is better.

File: ibis-server/rust/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "wren-core-wrapper"
```
Comment (goldmedal) [inline ibis-server/rust/Cargo.toml:2]:
```suggestion
name = "wren-core-python"
```
Maybe `wren-core-python` better.

**Review by goldmedal — APPROVED** [review]
Thanks @grieve54706. I think overall it looks good, just some minor comments. However, I believe you are working on another version for wren-core-python. If you want to merge this first or move on to the next version, either option is fine with me.

File: .github/workflows/ibis-ci.yml (hunk 3 above)
Comment (grieve54706) [reply .github/workflows/ibis-ci.yml:60]: Let's adjust it in the new PR to adjust the layout in the future.

File: ibis-server/rust/Cargo.toml (hunk 4 above)
Comment (grieve54706) [reply ibis-server/rust/Cargo.toml:2]: I will rename the name to `wren-modeling-py` to pair with `wren-modeling-rs`.

**Comment by grieve54706** [comment]
> Thanks @grieve54706. I think overall it looks good, just some minor comments. However, I believe you are working on another version for wren-core-python. If you want to merge this first or move on to the next version, either option is fine with me.

Let's merge this PR first. I will create another PR to adjust the layout to unmix the project.

**Pull request merged**